<launch>

    <arg name="robot_name" default="kinova_gen3" />
    <arg name="gripper_name" default="robotiq_2f_85" />
    <arg name="gamepad_name" default="logitech_f710" />
    <arg name="gamepad_device" default="js0" />

    <include file="$(find kortex_driver)/launch/kortex_driver.launch">
        <arg name="robot_name" value="$(arg robot_name)" />
        <arg name="gripper" value="$(arg gripper_name)" />
        <arg name="start_rviz" value="false" />
    </include>

    <node name="$(arg gamepad_name)_joy" pkg="joy" type="joy_node" ns="$(arg gamepad_name)">
        <param name="dev" value="/dev/input/$(arg gamepad_device)" />
    </node>

    <node name="joy2twist_node" pkg="joy2twist" type="joy2twist_node" output="screen">
        <rosparam file="$(find joy2twist)/config/joy2twist.yaml" />
        <remap from="joy" to="/$(arg gamepad_name)/joy" />
    </node>

    <group ns="$(arg robot_name)">
        <node name="kortex_joystick" pkg="kortex_joystick_demo" type="kortex_joystick_demo_node" output="screen">
            <rosparam file="$(find kortex_joystick_demo)/config/kortex_joystick_config.yaml" />
            <remap from="joy" to="/$(arg gamepad_name)/joy" />
        </node>
    </group>

</launch>